test/lib/restforce/db/association_cache_test.rb

```diff
@@ -0,0 +1,53 @@
+require_relative "../../../test_helper"
+
+describe Restforce::DB::AssociationCache do
+
+  configure!
+
+  let(:cache) { Restforce::DB::AssociationCache.new }
+  let(:database_model) { CustomObject }
+  let(:lookups) { { salesforce_id: "a001a000001E1vREAL" } }
+  let(:record) { database_model.new(lookups) }
+  let(:found) { cache.find(database_model, lookups) }
+
+  describe "#find" do
+
+    describe "when the record has been added to the cache" do
+      before do
+        cache << record
+      end
+
+      it "finds the record" do
+        expect(found).to_equal record
+      end
+    end
+
+    describe "when the record has been persisted" do
+      before do
+        record.save!
+      end
+
+      it "finds the persisted record by its lookups" do
+        expect(found).to_equal record
+      end
+    end
+
+    describe "when no record has been added or persisted" do
+
+      it "returns nil" do
+        expect(found).to_be_nil
+      end
+    end
+  end
+
+  describe "#<<" do
+    before do
+      cache << record
+    end
+
+    it "adds the record to the cache" do
+      expect(found).to_equal record
```

Comment:
Yeah, I was a little unsure how to reasonably test this without exposing pieces of API that really don't need to be exposed. I'll fidget with it and add some negative test cases.